CMakeLists.txt

```diff
@@ -192,6 +200,17 @@ if (ENABLE_MICROPROFILE)
             )
 endif (ENABLE_MICROPROFILE)
 
+# Everything in /test is a test source
+FILE(GLOB_RECURSE TEST_SOURCES test/*)
```

Comment:
Do not use globbing. This will be executed on configure time only and new files won't be detected.

Always specify source files in a list.

I suggest to add a CMakeLists.txt to test/ and just include it here.